### PR TITLE
Play ogg

### PR DESCRIPTION
**Pull request by joshuacox**
## Description
Adds a Ogg123Service and a play_ogg exactly like Mpg123Service and play_mp3

## How to test
I have skill for a podcast which does not have an mp3 feed:
https://github.com/joshuacox/skill-GNUworldOrder

## Contributor license agreement signed?
signed by @joshuacox

### CODE REVIEW

**Comment by joshuacox** [comment]
for now I've disabled the audio service to force play_ogg on that gnu world order skill, though I'm interested in making it work there as well.

**Comment by forslund** [comment]
This is a good addition. I'm wondering if the mpg123 and the ogg123 backend should be merged into a single *simple* backend

**Comment by joshuacox** [comment]
There is an ogg example backend in the config included now as well.   This is now working for me with the example skill:

https://github.com/joshuacox/skill-GNUworldOrder

I'm sure there is a more elegant way to handle both mp3 and oggs, and I'm certainly open to suggestions to better handling in either the skill or in the service.

**Comment by joshuacox** [comment]
Anything I can add to this PR to get it merged?  Although there might be more elegant ways to accomplish this, for now it is working, and without it my gnuworlorder skill does not work, which it just got covered in that podcast so there might be people attempting to get it to work.

**Comment by forslund** [comment]
Hi Josh, sorry for the delay. I've started to write a reply to you a couple of times but never finished.

In short I'd like to merge the ogg123 service and the mpg123 service into a "simple" backend.

For this I think we could
- rename the mpg123 backend folder to "simple"
- In the method handling playing add a check that checks the file-ending and/or MIME type of the file that should be played and select the appropriate binary to use for playing the file.

**Comment by forslund** [comment]
If you want me to help, build on this PR and add the changes I'd like to see.

**Comment by joshuacox** [comment]
I can add in a 'simple' service as well, but I'm not so certain it should be called 'simple'.  Perhaps 'auto' is more appropriate?  I'm certain this will fail on perfectly playable files if the MIME is wrong though, so perhaps it should be named with a warning like 'willFailOnWrongMimeType', I've thought about using the file extension as well, but the same stipulations apply.  Something like `file` in linux is much more reliable but not cross platform:
```
file /usr/share/sounds/KDE-Sys-File-Open-Foes.ogg                                                                                                                                                                                          
/usr/share/sounds/KDE-Sys-File-Open-Foes.ogg: Ogg data, Vorbis audio, stereo, 48000 Hz, ~192000 bps
```
Perhaps it should be in layers of fallback? file > mime type > extension > guess
I just added in the simple dir and pushed, I'm game on taking suggestions from you.

File: mycroft/audio/services/simple/__init__.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
```
[the remark this answers is not in the document]
Comment (joshuacox) [reply mycroft/audio/services/simple/__init__.py]: I think pep8 is crazy:

```
cat __init__.py|wc -l                  
126
cat __init__.py|awk 'length > 79'|wc -l
0
cat __init__.py|awk 'length < 79'|wc -l
126
```

**Comment by forslund** [comment]
I'm just guessing as I go along :)

The name `simple` is mainly an indication that it doesn't have all the bells and whistles like for example the `vlc` backend has. The mpg123 backend was mainly put in place to have a really basic backend that could play and queue mp3s.

My idea is that we do something simple. Use `mimetypes.guess_type()` to try to determine if it's an ogg or mp3. If no determination can be made just try to open it with one if that fails try the other. guess_type is rather stupid but is very fast.

In a second step we can add an option to specify mime type when sending a playlist to the audioservice. Basically allow for tuples `(filename, mime)` in the playlist. But this can be done in a second step.

File: mycroft/audio/services/simple/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuacox) [reply mycroft/audio/services/simple/__init__.py]: I think pep8 is concatenating those four lines

```
cat __init__.py|head -8|tail -4|wc -m
94
```

[the remark this answers is not in the document]
Comment (forslund) [reply mycroft/audio/services/simple/__init__.py]: Yeah, this looks weird

File: mycroft/audio/services/simple/__init__.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+from time import sleep
+
+from mycroft.audio.services import AudioBackend
+from mycroft.messagebus.message import Message
+from mycroft.util.log import LOG
+
+
+class SimpleAudioService(AudioBackend):
+    """
+        Simple Audio backend for both mpg123 and the ogg123 player. This one is rather limited and
```
Comment (forslund) [inline mycroft/audio/services/simple/__init__.py]: Travis complains of this line which is a bit wrong, try splitting this into two lines

**Comment by joshuacox** [comment]
this history is divergent (unrelated) to dev now, so abandoning this PR.  I will make two PRs one with the ogg123 player that  will only play oggs, but do so reliably and immediately for any listeners of the above podcast so that they will not see mycroft fail on the above skill and have their view tainted of mycroft overall.  

And after we take care of the above users, I will open a second PR where we can continue to work on this 'simple' player and hopefully automagically determine mime type (mimetypes.guess_type has performed very poorly at this task in my initial testing).

**Pull request closed without merging**